src/metabase/query_processor/middleware/cache_backend/db.clj

```diff
@@ -64,10 +60,9 @@
   ^PreparedStatement [strategy query-hash ^Connection conn]
   (if-not (:avg-execution-ms strategy)
     (log/debugf "Caching strategy %s needs :avg-execution-ms to work" (pr-str strategy))
-    (let [max-age-seconds (math/round (/ (* (:multiplier strategy)
-                                            (:avg-execution-ms strategy))
-                                         1000.0))]
-      (prepare-statement conn query-hash (seconds-ago max-age-seconds)))))
+    (let [max-age-ms (* (:multiplier strategy)
+                        (:avg-execution-ms strategy))]
+      (prepare-statement conn query-hash (t/max (ms-ago max-age-ms) (:invalidated-at strategy))))))
```

Comment:
`invalidated-at` here will be always `nil` in OSS version since it's introduced in `strategies.clj` (and that file is in enterprise part).